tools/SourceKit/include/SourceKit/Support/Concurrency.h

```diff
@@ -17,6 +17,55 @@
 
 namespace SourceKit {
 
+class Semaphore {
+public:
+  Semaphore(long count) {
+    ImplObj = Impl::create(count);
+  }
+  ~Semaphore() {
+    if (ImplObj)
+      Impl::release(ImplObj);
+  }
+
+  void signal() {
+    Impl::signal(ImplObj);
+  }
+
+  bool wait() {
+    return Impl::wait(ImplObj);
+  }
+
+  bool wait(long milliseconds) {
+    return Impl::wait(ImplObj, milliseconds);
+  }
+
+  Semaphore(const Semaphore &Other) {
```

Comment:
Sure, I do think that it makes sense to treat a semaphore as an object intended to be shared rather than copied. I think I was following the example of `WorkQueue` in this case.

I haven't written much C++ recently and am still trying to get a better understanding of which types of constructors are appropriate. What do you think about the move constructor here? Does it make sense as-is?